lib/reboot_cmds.py

```diff
@@ -41,6 +41,36 @@ def run_command(cmd, cfg):
         sys.exit(1)
 
 
+@assert_lock_file("ua-reboot-fix-pro-pkg-holds")
+def fix_pro_pkg_holds(args, cfg):
+    status_cache = cfg.read_cache("status-cache")
+    services_status_list = status_cache.get("services", [])
+    for service in services_status_list:
+        if service.get("name") in ("fips", "fips-updates"):
+            service_status = service.get("status")
+            if service_status == "enabled":
+                ent_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME[
+                    entitlement_name
+                ]
+                logging.debug(
+                    "Attempting to remove Ubuntu Pro FIPS package holds"
+                )
+                entitlement = ent_cls(cfg)
+                try:
+                    entitlement.setup_apt_config()  # Removes package holds
```

Comment:
I think we still need to manually install the `additionalPackages`  directive here, right ?

Maybe the `process_remaining_deltas`  function will cover that, but my concern is that if we update the contract on prod before the user upgrades the package, we may end up not installing the metapackage.

For example, consider this scenario:

1) fips contract is updated with new `additionalPackage (ubuntu-fips)`
2) User runs a `ua refresh`
3) User upgrade the `uaclient`

In that scenario we will not install the new `ubuntu-fips` metapackage, because there will be no deltas in the contract regarding it.

Reply:
+1 we can call entitlement.install_packages() here which will DTRT with regard to the cached additionalPackages directive on disk. 

for use cases:
 1. old client + new contract (via ua refresh): additionalPackage delta containing `ubuntu-fips` was ignored (so we need to install)
 2. new_client + new_contract from ua refresh: additionalPackages will attempt to install `ubuntu-fips` on additionalPackage delta so NOOP here if we try to re-install package that is already instaled
 3. plain upgrade to new_client, postinst will create the REBOOT_CMD_REQUIRED   marker and force this logic to run and correct new contract additionalPackages after reboot.


this means we have to release the contract update on PRO images before we release the new PRO images with version 26.0 support